File: backend/projectverseAPI/projectverseAPI/Models/Post.cs
namespace projectverseAPI.Models
{
    public class Post
    {
        public Guid Id { get; set; }
        public Project Project { get; set; }
        public Guid ProjectId { get; set; }
        public long ViewsCount { get; set; }
        public long LikesCount { get; set; }
        public IList<Like> Likes { get; set; }
        public IList<PostComment> PostComments { get; set; }
    }
}

File: frontend/projectverse/src/data/Post.ts
import PostComment from "./PostComments";
import { Project } from "./Profile";

interface Post{
  id:string,
  project:Project,
  viewsCount:number,
  likesCount:number,
  isLikedByCurrentUser:boolean,
  postComments:PostComment[]  
}

export default Post;